// pg-sdk-node/dist/common/events/builders/BaseEventBuilder.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseEventBuilder = void 0;
const BaseEvent_1 = require("../models/BaseEvent");
class BaseEventBuilder {
    constructor() {
        /**
         * SETTERS
         **/
        this.setMerchantOrderId = (merchantOrderId) => {
            this._merchantOrderId = merchantOrderId;
            return this;
        };
        this.setEventName = (eventName) => {
            this._eventName = eventName;
            return this;
        };
        this.setEventTime = (eventTime) => {
            this._eventTime = eventTime;
            return this;
        };
        this.setData = (data) => {
            this._data = data;
            return this;
        };
        this.build = () => {
            return new BaseEvent_1.BaseEvent(this._merchantOrderId, this._eventName, this._eventTime, this._data);
        };
    }
}
exports.BaseEventBuilder = BaseEventBuilder;
