// pg-sdk-node/dist/common/models/request/instruments/BillingAddress.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingAddress = void 0;
class BillingAddress {
    constructor(line1, line2, city, state, zip, country) {
        this.line1 = line1;
        this.line2 = line2;
        this.city = city;
        this.state = state;
        this.zip = zip;
        this.country = country;
    }
    static builder() {
        return new BillingAddressBuilder();
    }
}
exports.BillingAddress = BillingAddress;
class BillingAddressBuilder {
    constructor() {
        /**
         * SETTERS
         */
        this.line1 = (line1) => {
            this._line1 = line1;
        };
        this.line2 = (line2) => {
            this._line2 = line2;
        };
        this.city = (city) => {
            this._city = city;
        };
        this.state = (state) => {
            this._state = state;
        };
        this.zip = (zip) => {
            this._zip = zip;
        };
        this.country = (country) => {
            this._country = country;
        };
        this.build = () => {
            return new BillingAddress(this._line1, this._line2, this._city, this._state, this._zip, this._country);
        };
    }
}
