// pg-sdk-node/dist/common/models/request/instruments/CardPaymentV2Instrument.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardPaymentV2Instrument = void 0;
const PgV2InstrumentType_1 = require("../../PgV2InstrumentType");
const PaymentV2Instrument_1 = require("./PaymentV2Instrument");
class CardPaymentV2Instrument extends PaymentV2Instrument_1.PaymentV2Instrument {
    constructor(cardDetails, authMode, merchantUserId, savedCard) {
        super(PgV2InstrumentType_1.PgV2InstrumentType.CARD);
        this.authMode = authMode;
        this.cardDetails = cardDetails;
        this.merchantUserId = merchantUserId;
        this.savedCard = savedCard;
    }
}
exports.CardPaymentV2Instrument = CardPaymentV2Instrument;
CardPaymentV2Instrument.builder = () => {
    return new CardPaymentV2InstrumentBuilder();
};
class CardPaymentV2InstrumentBuilder {
    constructor() {
        this.cardDetails = (cardDetails) => {
            this._cardDetails = cardDetails;
            return this;
        };
        this.authMode = (authMode) => {
            this._authMode = authMode;
            return this;
        };
        this.savedCard = (savedCard) => {
            this._savedCard = savedCard;
            return this;
        };
        this.merchantUserId = (merchantUserId) => {
            this._merchantUserId = merchantUserId;
            return this;
        };
        this.build = () => {
            return new CardPaymentV2Instrument(this._cardDetails, this._authMode, this._merchantUserId, this._savedCard);
        };
    }
}
