// pg-sdk-node/dist/common/models/request/instruments/Expiry.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Expiry = void 0;
class Expiry {
    constructor(month, year) {
        this.month = month;
        this.year = year;
    }
}
exports.Expiry = Expiry;
Expiry.builder = () => {
    return new ExpiryBuilder();
};
class ExpiryBuilder {
    constructor() {
        this.expiryMonth = (expiryMonth) => {
            this._expiryMonth = expiryMonth;
            return this;
        };
        this.expiryYear = (expiryYear) => {
            this._expiryYear = expiryYear;
            return this;
        };
        this.build = () => {
            return new Expiry(this._expiryMonth, this._expiryYear);
        };
    }
}
