// pg-sdk-node/dist/common/models/request/instruments/NewCardDetails.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewCardDetails = void 0;
class NewCardDetails {
    constructor(encryptedCardNumber, encryptedCvv, encryptionKeyId, expiry, cardHolderName) {
        this.encryptedCardNumber = encryptedCardNumber;
        this.encryptedCvv = encryptedCvv;
        this.encryptionKeyId = encryptionKeyId;
        this.expiry = expiry;
        this.cardHolderName = cardHolderName;
    }
}
exports.NewCardDetails = NewCardDetails;
NewCardDetails.builder = () => {
    return new NewCardDetailsBuilder();
};
class NewCardDetailsBuilder {
    constructor() {
        /**
         * SETTERS
         */
        this.encryptedCardNumber = (encryptedCardNumber) => {
            this._encryptedCardNumber = encryptedCardNumber;
            return this;
        };
        this.encryptedCvv = (encryptedCvv) => {
            this._encryptedCvv = encryptedCvv;
            return this;
        };
        this.encryptionKeyId = (encryptionKeyId) => {
            this._encryptionKeyId = encryptionKeyId;
            return this;
        };
        this.expiry = (expiry) => {
            this._expiry = expiry;
            return this;
        };
        this.cardHolderName = (cardHolderName) => {
            this._cardHolderName = cardHolderName;
            return this;
        };
        this.build = () => {
            return new NewCardDetails(this._encryptedCardNumber, this._encryptedCvv, this._encryptionKeyId, this._expiry, this._cardHolderName);
        };
    }
}
