// pg-sdk-node/dist/common/models/request/instruments/TokenDetails.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenDetails = void 0;
class TokenDetails {
    constructor(encryptedToken, encryptedCvv, encryptionKeyId, expiry, cryptogram, panSuffix, cardHolderName, eci, atc) {
        this.encryptedToken = encryptedToken;
        this.encryptedCvv = encryptedCvv;
        this.encryptionKeyId = encryptionKeyId;
        this.expiry = expiry;
        this.cryptogram = cryptogram;
        this.panSuffix = panSuffix;
        this.cardHolderName = cardHolderName;
        this.eci = eci;
        this.atc = atc;
    }
}
exports.TokenDetails = TokenDetails;
TokenDetails.builder = () => {
    return new TokenDetailsBuilder();
};
class TokenDetailsBuilder {
    constructor() {
        /**
         * SETTERS
         */
        this.encryptedToken = (encryptedToken) => {
            this._encryptedToken = encryptedToken;
            return this;
        };
        this.encryptedCvv = (encryptedCvv) => {
            this._encryptedCvv = encryptedCvv;
            return this;
        };
        this.encryptionKeyId = (encryptionKeyId) => {
            this._encryptionKeyId = encryptionKeyId;
            return this;
        };
        this.expiry = (expiry) => {
            this._expiry = expiry;
            return this;
        };
        this.cryptogram = (cryptogram) => {
            this._cryptogram = cryptogram;
            return this;
        };
        this.panSuffix = (panSuffix) => {
            this._panSuffix = panSuffix;
            return this;
        };
        this.cardHolderName = (cardHolderName) => {
            this._cardHolderName = cardHolderName;
            return this;
        };
        this.eci = (eci) => {
            this._eci = eci;
            return this;
        };
        this.atc = (atc) => {
            this._atc = atc;
            return this;
        };
        this.build = () => {
            return new TokenDetails(this._encryptedToken, this._encryptedCvv, this._encryptionKeyId, this._expiry, this._cryptogram, this._panSuffix, this._cardHolderName, this._eci, this._atc);
        };
    }
}
