// pg-sdk-node/dist/common/models/request/instruments/UpiCollectPayViaPhoneNumberRequestBuilder.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpiCollectPayViaPhoneNumberRequestBuilder = void 0;
const CustomCheckoutPayRequest_1 = require("../../../../payments/v2/models/request/CustomCheckoutPayRequest");
const CollectPaymentV2Instrument_1 = require("./CollectPaymentV2Instrument");
const PhoneNumberCollectPaymentDetails_1 = require("./PhoneNumberCollectPaymentDetails");
const PgPaymentFlow_1 = require("../../../../payments/v2/models/request/PgPaymentFlow");
class UpiCollectPayViaPhoneNumberRequestBuilder {
    constructor() {
        /**
         * SETTERS
         */
        this.merchantOrderId = (merchantOrderId) => {
            this._merchantOrderId = merchantOrderId;
            return this;
        };
        this.amount = (amount) => {
            this._amount = amount;
            return this;
        };
        this.metaInfo = (metaInfo) => {
            this._metaInfo = metaInfo;
            return this;
        };
        this.constraints = (constraints) => {
            this._constraints = constraints;
            return this;
        };
        this.message = (message) => {
            this._message = message;
            return this;
        };
        this.phoneNumber = (phoneNumber) => {
            this._phoneNumber = phoneNumber;
            return this;
        };
        this.expireAfter = (expireAfter) => {
            this._expireAfter = expireAfter;
            return this;
        };
        this.build = () => {
            const paymentFlow = PgPaymentFlow_1.PgPaymentFlow.builder()
                .paymentMode(CollectPaymentV2Instrument_1.CollectPaymentV2Instrument.builder()
                .details(PhoneNumberCollectPaymentDetails_1.PhoneNumberCollectPaymentDetails.builder()
                .phoneNumber(this._phoneNumber)
                .build())
                .message(this._message)
                .build())
                .build();
            return new CustomCheckoutPayRequest_1.CustomCheckoutPayRequest(this._merchantOrderId, this._amount, paymentFlow, this._expireAfter, this._metaInfo, this._constraints);
        };
    }
}
exports.UpiCollectPayViaPhoneNumberRequestBuilder = UpiCollectPayViaPhoneNumberRequestBuilder;
