// pg-sdk-node/dist/common/events/models/BulkEvent.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkEvent = void 0;
const Headers_1 = require("../../constants/Headers");
const Constants_1 = require("../Constants");
class BulkEvent {
    constructor(events, source, clientVersion) {
        this.source = Constants_1.Constants.SOURCE;
        this.clientVersion = Headers_1.Headers.SDK_TYPE + ':' + Headers_1.Headers.SDK_VERSION;
        this.events = events;
        this.source = source;
        this.clientVersion = clientVersion;
    }
}
exports.BulkEvent = BulkEvent;
BulkEvent.builder = () => {
    return new BulkEventBuilder();
};
class BulkEventBuilder {
    constructor() {
        this.events = (eventList) => {
            this._events = eventList;
            return this;
        };
        this.source = (source) => {
            this._source = source;
            return this;
        };
        this.clientVersion = (clientVersion) => {
            this._clientVersion = clientVersion;
            return this;
        };
        this.build = () => {
            var _a, _b;
            this._source = (_a = this._source) !== null && _a !== void 0 ? _a : Constants_1.Constants.SOURCE;
            this._clientVersion =
                (_b = this._clientVersion) !== null && _b !== void 0 ? _b : Headers_1.Headers.SDK_TYPE + ':' + Headers_1.Headers.SDK_VERSION;
            return new BulkEvent(this._events, this._source, this._clientVersion);
        };
    }
}
