// pg-sdk-node/dist/common/configs/CredentialConfig.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CredentialConfigBuilder = exports.CredentialConfig = void 0;
class CredentialConfig {
    constructor(clientId, clientSecret, clientVersion) {
        this._clientId = clientId;
        this._clientSecret = clientSecret;
        this._clientVersion = clientVersion;
    }
    static builder() {
        return new CredentialConfigBuilder();
    }
    /**
     * Getter Functions
     */
    get clientId() {
        return this._clientId;
    }
    get clientSecret() {
        return this._clientSecret;
    }
    get clientVersion() {
        return this._clientVersion;
    }
}
exports.CredentialConfig = CredentialConfig;
class CredentialConfigBuilder {
    constructor() {
        /**
         * Setters used for MerchantConfig builder
         */
        this.clientId = (clientId) => {
            this._clientId = clientId;
            return this;
        };
        this.clientSecret = (clientSecret) => {
            this._clientSecret = clientSecret;
            return this;
        };
        this.clientVersion = (clientVersion) => {
            this._clientVersion = clientVersion;
            return this;
        };
        this.build = () => {
            return new CredentialConfig(this._clientId, this._clientSecret, this._clientVersion);
        };
    }
}
exports.CredentialConfigBuilder = CredentialConfigBuilder;
