// pg-sdk-node/dist/common/models/PgV2InstrumentType.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PgV2InstrumentType = void 0;
var PgV2InstrumentType;
(function (PgV2InstrumentType) {
    PgV2InstrumentType["UPI_COLLECT"] = "UPI_COLLECT";
    PgV2InstrumentType["UPI_INTENT"] = "UPI_INTENT";
    PgV2InstrumentType["PPE_INTENT"] = "PPE_INTENT";
    PgV2InstrumentType["UPI_QR"] = "UPI_QR";
    PgV2InstrumentType["CARD"] = "CARD";
    PgV2InstrumentType["TOKEN"] = "TOKEN";
    PgV2InstrumentType["NET_BANKING"] = "NET_BANKING";
})(PgV2InstrumentType = exports.PgV2InstrumentType || (exports.PgV2InstrumentType = {}));
