// pg-sdk-node/dist/common/models/response/CallbackType.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallbackType = void 0;
var CallbackType;
(function (CallbackType) {
    CallbackType[CallbackType["PG_ORDER_COMPLETED"] = 0] = "PG_ORDER_COMPLETED";
    CallbackType[CallbackType["PG_ORDER_FAILED"] = 1] = "PG_ORDER_FAILED";
    CallbackType[CallbackType["PG_TRANSACTION_ATTEMPT_FAILED"] = 2] = "PG_TRANSACTION_ATTEMPT_FAILED";
    CallbackType[CallbackType["PG_REFUND_COMPLETED"] = 3] = "PG_REFUND_COMPLETED";
    CallbackType[CallbackType["PG_REFUND_FAILED"] = 4] = "PG_REFUND_FAILED";
    CallbackType[CallbackType["PG_REFUND_ACCEPTED"] = 5] = "PG_REFUND_ACCEPTED";
    CallbackType[CallbackType["CHECKOUT_ORDER_COMPLETED"] = 6] = "CHECKOUT_ORDER_COMPLETED";
    CallbackType[CallbackType["CHECKOUT_ORDER_FAILED"] = 7] = "CHECKOUT_ORDER_FAILED";
    CallbackType[CallbackType["CHECKOUT_TRANSACTION_ATTEMPT_FAILED"] = 8] = "CHECKOUT_TRANSACTION_ATTEMPT_FAILED";
})(CallbackType = exports.CallbackType || (exports.CallbackType = {}));
