// pg-sdk-node/dist/common/models/request/DeviceContext.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceContext = void 0;
class DeviceContext {
    constructor(deviceOS, merchantCallBackScheme) {
        this.deviceOS = deviceOS;
        this.merchantCallBackScheme = merchantCallBackScheme;
    }
    static builder() {
        return new DeviceContextBuilder();
    }
}
exports.DeviceContext = DeviceContext;
class DeviceContextBuilder {
    constructor() {
        /**
         * SETTERS
         */
        this.deviceOS = (deviceOS) => {
            this._deviveOS = deviceOS;
            return this;
        };
        this.merchantCallBackScheme = (merchantCallBackScheme) => {
            this._merchantCallBackScheme = merchantCallBackScheme;
            return this;
        };
        this.build = () => {
            return new DeviceContext(this._deviveOS, this._merchantCallBackScheme);
        };
    }
}
