// pg-sdk-node/dist/common/models/request/RefundRequest.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefundRequest = void 0;
/**
 * Creates a request to initiate a refund -> RefundRequest.builder()
 */
class RefundRequest {
    constructor(merchantRefundId, originalMerchantOrderId, amount) {
        this.merchantRefundId = merchantRefundId;
        this.originalMerchantOrderId = originalMerchantOrderId;
        this.amount = amount;
    }
}
exports.RefundRequest = RefundRequest;
RefundRequest.builder = () => {
    return new RefundRequestBuilder();
};
class RefundRequestBuilder {
    constructor() {
        this.merchantRefundId = (merchantRefundId) => {
            this._merchantRefundId = merchantRefundId;
            return this;
        };
        this.originalMerchantOrderId = (originalMerchantOrderId) => {
            this._originalMerchantOrderId = originalMerchantOrderId;
            return this;
        };
        this.amount = (amount) => {
            this._amount = amount;
            return this;
        };
        this.build = () => {
            return new RefundRequest(this._merchantRefundId, this._originalMerchantOrderId, this._amount);
        };
    }
}
