// pg-sdk-node/dist/payments/v2/models/request/CreateSdkOrderRequest.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSdkOrderRequest = void 0;
const PgCheckoutPaymentFlow_1 = require("./PgCheckoutPaymentFlow");
/**
 * Creates a request to initiate a order for SDK Integration
 * 1. For Standard Checkout -> CreateSdkOrderRequest.StandardCheckoutBuilder()
 * 2. For Custom Checkout -> CreateSdkOrderRequest.CustomCheckoutBuilder()
 */
class CreateSdkOrderRequest {
    constructor(merchantOrderId, amount, paymentFlow, metaInfo, expireAfter, constraints, disablePaymentRetry) {
        this.merchantOrderId = merchantOrderId;
        this.amount = amount;
        this.metaInfo = metaInfo;
        this.paymentFlow = paymentFlow;
        this.constraints = constraints;
        this.expireAfter = expireAfter;
        this.disablePaymentRetry = disablePaymentRetry;
    }
}
exports.CreateSdkOrderRequest = CreateSdkOrderRequest;
CreateSdkOrderRequest.StandardCheckoutBuilder = () => {
    return new StandardCheckoutBuilder();
};
CreateSdkOrderRequest.CustomCheckoutBuilder = () => {
    return new CustomCheckoutBuilder();
};
class CustomCheckoutBuilder {
    constructor() {
        /**
         * SETTERS
         */
        this.merchantOrderId = (merchantOrderId) => {
            this._merchantOrderId = merchantOrderId;
            return this;
        };
        this.amount = (amount) => {
            this._amount = amount;
            return this;
        };
        this.metaInfo = (metaInfo) => {
            this._metaInfo = metaInfo;
            return this;
        };
        this.message = (message) => {
            this._message = message;
            return this;
        };
        this.redirectUrl = (redirectUrl) => {
            this._redirectUrl = redirectUrl;
            return this;
        };
        this.constraints = (constraints) => {
            this._constraints = constraints;
            return this;
        };
        this.expireAfter = (expireAfter) => {
            this._expireAfter = expireAfter;
            return this;
        };
        this.disablePaymentRetry = (disablePaymentRetry) => {
            this._disablePaymentRetry = disablePaymentRetry;
            return this;
        };
        this.build = () => {
            const paymentFlow = PgCheckoutPaymentFlow_1.PgCheckoutPaymentFlow.builder()
                .redirectUrl(this._redirectUrl)
                .message(this._message)
                .build();
            return new CreateSdkOrderRequest(this._merchantOrderId, this._amount, paymentFlow, this._metaInfo, this._expireAfter, this._constraints, this._disablePaymentRetry);
        };
    }
}
class StandardCheckoutBuilder {
    constructor() {
        /**
         * SETTERS
         */
        this.merchantOrderId = (merchantOrderId) => {
            this._merchantOrderId = merchantOrderId;
            return this;
        };
        this.amount = (amount) => {
            this._amount = amount;
            return this;
        };
        this.metaInfo = (metaInfo) => {
            this._metaInfo = metaInfo;
            return this;
        };
        this.message = (message) => {
            this._message = message;
            return this;
        };
        this.redirectUrl = (redirectUrl) => {
            this._redirectUrl = redirectUrl;
            return this;
        };
        this.expireAfter = (expireAfter) => {
            this._expireAfter = expireAfter;
            return this;
        };
        this.disablePaymentRetry = (disablePaymentRetry) => {
            this._disablePaymentRetry = disablePaymentRetry;
            return this;
        };
        this.build = () => {
            const paymentFlow = PgCheckoutPaymentFlow_1.PgCheckoutPaymentFlow.builder()
                .redirectUrl(this._redirectUrl)
                .message(this._message)
                .build();
            return new CreateSdkOrderRequest(this._merchantOrderId, this._amount, paymentFlow, this._metaInfo, this._expireAfter, undefined, this._disablePaymentRetry);
        };
    }
}
