// pg-sdk-node/dist/payments/v2/models/request/PgCheckoutPaymentFlow.js
"use strict";
/*
 *  Copyright (c) 2025 Original Author(s), PhonePe India Pvt. Ltd.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PgCheckoutPaymentFlow = void 0;
const MerchantUrls_1 = require("./MerchantUrls");
const PaymentFlow_1 = require("../../../../common/models/PaymentFlow");
const PaymentFlowType_1 = require("../../../../common/models/PaymentFlowType");
class PgCheckoutPaymentFlow extends PaymentFlow_1.PaymentFlow {
    constructor(message, redirectUrl) {
        super(PaymentFlowType_1.PaymentFlowType.PG_CHECKOUT);
        this.message = message;
        this.merchantUrls = new MerchantUrls_1.MerchantUrls(redirectUrl);
    }
    static builder() {
        return new PgCheckoutPaymentFlowBuilder();
    }
}
exports.PgCheckoutPaymentFlow = PgCheckoutPaymentFlow;
class PgCheckoutPaymentFlowBuilder {
    /**
     * SETTERS FOR PG_CHECKOUT PAYMENT FLOW
     */
    message(message) {
        this._message = message;
        return this;
    }
    redirectUrl(redirectUrl) {
        this._redirectUrl = redirectUrl;
        return this;
    }
    build() {
        return new PgCheckoutPaymentFlow(this._message, this._redirectUrl);
    }
}
